javascript/modules/client/src/client-addressList.ts

```diff
@@ -350,6 +377,36 @@ export class ClientAddressList extends ClientCore
     };
   }
 
+  private async *_canVote(
+    params: ICanVoteParams,
+  ): AsyncGenerator<CanVoteStepValue> {
+    const signer = this.web3.getConnectedSigner();
+    if (!signer) {
+      throw new NoSignerError();
+    } else if (!signer.provider) {
+      throw new NoProviderError();
+    }
+    if (!isAddress(params.address) || !isAddress(params.pluginAddress)) {
+      throw new InvalidAddressError();
+    }
+    if (!isProposalId(params.proposalId)) {
+      throw new InvalidProposalIdError();
+    }
+
+    // TODO: Implement
+    await delay(1000);
+    yield {
+      key: CanVoteStep.CHECKING,
+      txHash:
+        "0x0123456789012345678901234567890123456789012345678901234567890123",
+    };
+    await delay(3000);
+    yield {
+      key: CanVoteStep.DONE,
+      canVote: parseInt(params.address.slice(-1), 16) % 2 === 1,
+    };
```

Comment:
This should be a call to Subgraph or Ethereum, there is no need for a signer or a transaction